test/controllers/deprecated_secrets_controller_test.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class DeprecatedSecretsControllerTest < ActionDispatch::IntegrationTest
+  def create_session_for(user)
+    Passwordless::Session.create!(
+      authenticatable: user,
+      remote_addr: "yes",
+      user_agent: "James Bond"
+    )
+  end
+
+  def cookie_key(authenticatable_class)
+    DeprecatedSecretsController.new.send(:cookie_key, authenticatable_class)
```

Comment:
Let's pull this method out into its own file so we can test it on its own and we don't need weird (in a Rails context) initialisation?